### DISCUSSION TITLE
Option to sync velocity with NetworkTransform.

### DISCUSSION THREAD

**Discussion by AlphaLul** [FirstGearGames/FishNet#451] [Feature Request]
For some games with movement where players switch directions quickly, predicting via rigidbody can result in jittery movement, making it better to use a NetworkTransform instead. However, if you want to have predicted/client-side forces on remote clients, you need to synchronize the velocity. It would be nice to have an option to synchronize the velocity of your rigidbody/rigidbody2d to make this easier.

**Comment by FirstGearGames** [comment]
These two are not the same thing. Synchronizing velocities on a NetworkTransform will still result in jitter on collision and massive desyncs.

**Reply by AlphaLul** [reply]
I remember seeing a plan for synced velocity or something of the sort on the roadmap before it was moved here. What was the plan for it back then?

For my game, there are black-hole-type gravitational fields that players can get sucked into and slowly move out of by moving away from the center. They work on every client so predicted clients behave correctly. When other clients go into the field and get sucked in, they can't move out from the perspective of the local client because inputs/velocity isn't synced, only transform, so they get teleported slightly out and get sucked back in over and over until they're completely out.

I would use prediction v2 for this but it makes my movement pretty jittery for some reason, and also since my players change directions really quickly, it always results in a false prediction when they change directions. I figured that just syncing velocity in these cases could work instead, but if not, are there any other options? Or will prediction v2 be better at handling quick changes in movement in the future? I would actually like to see an option where inputs are still forwarded, but they aren't used for prediction, just applied when the movements would have been sent by the server anyways.

**Reply by FirstGearGames** [reply]
Essentially, it's pointless. If the object has velocity it will disrupt the smoothing on the NetworkTransform. I know assets like SmoothSync does it, but it isn't clean and will absolutely cause accuracy issues.

Your best bet is to hook into the OnNextGoal, pull the move rates from it, and set those as your velocities.
I'm going to close this out but I'm open to further discussion on my Discord. Let me know!
